api/api.go

```diff
@@ -219,12 +229,29 @@ func (b *BlockChainAPI) GetBalance(
 		return handleError[*hexutil.Big](err, l, b.collector)
 	}
 
+	cacheKey := getCacheKey(address, evmHeight)
```

Comment:
The CI is failing because of the special values for `evmHeight`. For example:
```json
{
  "jsonrpc": "2.0",
  "method": "eth_getBalance",
  "params": [
    "0xa94f5374Fce5edBC8E2a8697C15331677e6EbF0B",
    "latest"
  ],
  "id": 10
}
```
This is a very common JSON-RPC request for retrieving the balance of an address. The `"latest"` special value is used to indicate the latest EVM block height. In the code, this means that `evmHeight` has the special value `-2`. Internally, we use this special value to call `ExecuteScriptAtLatestBlock` instead of `ExecuteScriptAtBlockHeight`.

We might want to do something like this:
```go
var height int64
if evmHeight < 0 {
	latestEVMHeight, err := b.blocks.LatestEVMHeight()
	if err != nil {
		return handleError[*hexutil.Big](err, l, b.collector)
	}
	height = int64(latestEVMHeight)
} else {
	height = evmHeight
}
cacheKey := getCacheKey(address, height)
```
to translate the special value `-2` to an actual EVM block height.